utils/key/key.go

```diff
@@ -3,20 +3,44 @@ package key
 import (
 	"bytes"
 	"fmt"
+	"golang.org/x/crypto/sha3"
 	"strings"
 
 	"golang.org/x/exp/constraints"
 
 	"github.com/axelarnetwork/utils/convert"
 )
 
+var usedPrefixes = map[prefixID]struct{}{}
+
+type prefixID struct {
+	module string
```

Comment:
Don't dislike the idea of coupling it with the module, but this would prevent the evm's BaseKeeper and ChainKeeper to use overlapping prefixes when in reality it's perfectly ok.

Reply:
I like that namespace better. Modules can have multiple namespaces for good reasons.